src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/VisitSessionAdminController.kt

```diff
@@ -71,28 +76,23 @@ class VisitSessionAdminController(
       example = "MDI",
     )
     prisonCode: String,
-    @RequestParam(value = "dayOfWeek", required = false)
-    @Parameter(
-      description = "Filter results by day of week",
-      example = "MONDAY",
-    )
-    dayOfWeek: DayOfWeek?,
-    @RequestParam(value = "validFrom", required = false)
-    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
-    @Parameter(
-      description = "Filter results by that when the session template is valid from",
-      example = "2021-11-03",
-    )
-    rangeStartDate: LocalDate?,
-    @RequestParam(value = "validTo", required = false)
+    @RequestParam(value = "rangeType", required = true)
     @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
     @Parameter(
-      description = "Filter results by that when the session template is valid to",
-      example = "2021-11-03",
+      description = "Filters session templates depending on there from and to Date",
+      example = "ACTIVE_OR_FUTURE",
     )
-    rangeEndDate: LocalDate?,
+    rangeType: SessionTemplateRangeType,
   ): List<SessionTemplateDto> {
-    return sessionTemplateService.getSessionTemplates(prisonCode, dayOfWeek, rangeStartDate, rangeEndDate)
+    var fromDate: LocalDate? = null
+    var toDate: LocalDate? = null
+    if (ACTIVE_OR_FUTURE == rangeType) {
+      fromDate = LocalDate.now()
+    }
+    if (HISTORIC == rangeType) {
```

Comment:
If choosing `HISTORIC`, what should I expect? Should it be templates that are no longer active at all?

Having tested, it is returning templates where the **from/to** dates are **both** in the past. But it is also returning templates that don't have a `dateTo` set at all - i.e. their `fromDate` is in the past but they are still active.

Reply:
@tpmcgowan Yes you would get one's that are no longer valid